digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala

```diff
@@ -0,0 +1,79 @@
+package fi.liikennevirasto.viite
+
+import java.sql.SQLIntegrityConstraintViolationException
+
+import fi.liikennevirasto.digiroad2.{DigiroadEventBus, GeometryUtils}
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.viite.AddressConsistencyValidator.AddressError.{InconsistentTopology, OverlappingRoadAddresses}
+import fi.liikennevirasto.viite.dao._
+import org.slf4j.LoggerFactory
+
+
+class RoadNetworkService(eventbus: DigiroadEventBus) {
+
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+
+  def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+
+  val logger = LoggerFactory.getLogger(getClass)
+
+  def checkRoadAddressNetwork(roadNumber: Option[Long]): Unit = {
+
+    def checkCalibrationPoints(road1: RoadAddress, road2: RoadAddress): Boolean = {
+      road1.track != road2.track && road1.calibrationPoints._1.isEmpty && road1.calibrationPoints._2.isEmpty &&
+        road2.calibrationPoints._1.isEmpty && road2.calibrationPoints._2.isEmpty
+    }
+
+    def checkOverlapping(road1: RoadAddress, road2: RoadAddress) = {
+      road1.endAddrMValue != road2.startAddrMValue && road1.track.value == road2.track.value match {
```

Comment:
Another question: Does could this benefit from the overlaps method on the GeometryUtils or is it a completely different purpose?